internal/publisher/core.go

```diff
@@ -0,0 +1,59 @@
+package publisher
+
+import (
+	"context"
+	"fmt"
+
+	"github.com/golang/protobuf/proto"
+	"github.com/razorpay/metro/internal/brokerstore"
+	"github.com/razorpay/metro/pkg/logger"
+	"github.com/razorpay/metro/pkg/messagebroker"
+	metrov1 "github.com/razorpay/metro/rpc/proto/v1"
+)
+
+// Core implements IPublisher
+type Core struct {
+	bs brokerstore.IBrokerStore
+}
+
+// NewCore returns a new publisher
+func NewCore(bs brokerstore.IBrokerStore) *Core {
+	return &Core{bs}
+}
+
+// Publish messages
+func (p *Core) Publish(ctx context.Context, req *metrov1.PublishRequest) ([]string, error) {
+	producer, err := p.bs.GetOrCreateProducer(ctx, messagebroker.ProducerClientOptions{Topic: req.Topic, TimeoutSec: 50})
+	if err != nil {
+		logger.Ctx(ctx).Errorw("error in getting producer", "msg", err.Error())
+		return nil, err
+	}
+
+	msgIDs := make([]string, 0)
+
+	for _, msg := range req.Messages {
+		// unset message id and publishtime if set
+		msg.MessageId = ""
```

Comment:
do we allow setting these in future ? if not, should we validate this and dont allow setting them at all

Reply:
Goofle pubsub doesn't throw an error here and silently ignores the user set message id.